platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileReader.java

```diff
@@ -73,8 +73,10 @@ private SignedStateFileReader() {}
                     return in.readSerializable();
                 });
 
-        final MerkleNodeState merkleNodeState =
-                initializeMerkleNodeState(createStateFromVirtualMap, data.stateRoot(), platformContext.getMetrics());
+        final VirtualMap virtualMap = (VirtualMap) data.stateRoot();
```

Comment:
Do I get it right that this code is used for all apps, including testing apps that still use MerkleStateRoot? If so, all testing app snapshots would be broken here, their state roots aren't virtual maps